allenact_plugins/lighthouse_plugin/lighthouse_tasks.py

```diff
@@ -276,20 +284,27 @@ def query_expert(self, **kwargs) -> Tuple[Any, bool]:
             else:
                 expert_action = last_action
 
-            return (
-                np.array(
-                    [
-                        expert_action == d,
-                        expert_action == r,
-                        expert_action == u,
-                        expert_action == l,
-                    ]
-                ),
-                True,
+            policy = np.array(
+                [
+                    expert_action == d,
+                    expert_action == r,
+                    expert_action == u,
+                    expert_action == l,
+                ]
             )
         else:
             raise NotImplementedError("Can only query expert for world dims of 1 or 2.")
 
+        if return_policy:
+            return policy, True
+        elif deterministic:
+            return int(np.argmax(policy)), True
+        else:
+            return (
```

Comment:
Is this a noisy expert?

Reply:
It's an expert where several actions are equally good at certain timesteps and so, previously, I'd returned a policy that put equal weight on all of these policies. I'm trying to port some old code to use AllenAct >=0.3 and, as the expert policy loss fixes are still pending, I added this to allow for deterministically returning actions rather than policies.